component/prometheus-operator.jsonnet

```
@@ -0,0 +1,23 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local inv = kap.inventory();
+local com = import 'lib/commodore.libjsonnet';
+// The hiera parameters for the component
+local params = inv.parameters.kube_prometheus;
+
+local kp =
+  (import 'kube-prometheus/main.libsonnet') {
+    values+:: {
+      common+: {
+        namespace: params.namespace,
+      },
+    },
+
+    prometheusOperator+: com.makeMergeable(params.prometheus_operator.params),
+  };
+
+{
+  ['10_prometheus-operator-' + name]: kp.prometheusOperator[name]
+  for name in std.filter((function(name) name != 'serviceMonitor' && name != 'prometheusRule'), std.objectFields(kp.prometheusOperator))
+} +
+{ '10_prometheus-operator-serviceMonitor': kp.prometheusOperator.serviceMonitor } +
+{ '10_prometheus-operator-prometheusRule': kp.prometheusOperator.prometheusRule }
```

Why do you even do this? Isn't this exactly the same as just having

```suggestion
{
  ['10_prometheus-operator-' + name]: kp.prometheusOperator[name]
  for name in std.objectFields(kp.prometheusOperator)
}
```

---

This is copy/pasta from upstream: https://github.com/prometheus-operator/kube-prometheus#compiling

The comment there says:

>serviceMonitor and prometheusRule are separated so that they can be created after the CRDs are ready

---

Doesn't matter since we deploy everything with ArgoCD, which is smart enough to create CRDs first when they're part of the same application.